src/libraries/Microsoft.Bcl.TimeProvider/src/System/Threading/Tasks/TimeProviderTaskExtensions.cs

```diff
@@ -0,0 +1,171 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Threading.Tasks
+{
+    public static class TimeProviderTaskExtensions
+    {
+        private sealed class DelayState : TaskCompletionSource<bool>
+        {
+            public DelayState() : base(TaskCreationOptions.RunContinuationsAsynchronously) {}
+            public ITimer Timer { get; set; }
+            public CancellationTokenRegistration Registration { get; set; }
+        }
+
+        public static Task Delay(this TimeProvider timeProvider, TimeSpan delay, CancellationToken cancellationToken = default)
```

Comment:
When building for .NET 8, the entire implementation of this should just be delegating to Task.Delay(delay, timeProvider, cancellationToken).

Reply:
This file is not built on .NET 8.0. It is only supported on the down-level. You may look at the csproj for that logic.

Reply:
That doesn't sound right.  You're saying someone could be referencing this library, building for netstandard2.0, and then when they change their project to target net8.0 it fails to build because net8.0 doesn't provide this timeProvider.Delay API?  The library should be exposing the same surface area regardless of its target.

Reply:
I'll fix that.

Reply:
Glad to see that API Compat caught this: https://github.com/dotnet/runtime/runs/12463425396
```
.packages\microsoft.dotnet.apicompat.task\8.0.100-preview.2.23107.1\build\Microsoft.NET.ApiCompat.ValidatePackage.targets(39,5): error CP0001: (NETCORE_ENGINEERING_TELEMETRY=Build) Type 'System.Threading.Tasks.TimeProviderTaskExtensions' exists on lib/netstandard2.0/Microsoft.Bcl.TimeProvider.dll but not on lib/net8.0/Microsoft.Bcl.TimeProvider.dll
```

If you want to repro locally in the future, just run `dotnet pack` on the csproj.

Reply:
Yup. we are clean now :-)